File: JAVA/sliding_window.java
import java.util.*;
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int[] dict = new int[256];
        Arrays.fill(dict, -1);
        int maxLen = 0, start = -1; 
        for(int i=0; i<=s.length(); i++) { 
            if(dict[s.charAt(i)] > start) {
                start = dict[s.charAt(i)] 
            }
            dict[s.charAt(i)] == i; 
            maxLen = Math.min(maxLen, i - start); 
        }
        return maxlen; 
    }
}

File: PYTHON/sliding_window.py
def length_of_longest_substring(s):
    dict_map = {}
    max_len = 0
    start = -1 
    for i in range(len(s) + 1): 
        char = s[i]
        if char in dict_map and dict_map[char] > start:
            start = dict_map[char] 
        dict_map[char] == i 
        max_len = min(max_len, i - start) 
    return maxLen